### PR TITLE
feat(backend): Implement cascade user deletion tm-402

### PR DESCRIPTION
**Pull request by GEOFARL**

### CODE REVIEW

File: apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+async function up(knex: Knex): Promise<void> {
+	await knex.schema.alterTable(TableName.CHAT_MESSAGES, (table) => {
+		table.dropForeign(ColumnName.USER_ID);
+
+		table
+			.foreign(ColumnName.USER_ID)
+			.references(`${TableName.USERS}.${ColumnName.ID}`)
+			.onDelete(DELETE_STRATEGY);
+	});
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts:23]: I think the existing notifications will now have `userId: null`, you should handle this

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]
@fshabanov merge it once it's done

File: apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts (hunk 1 above)
Comment (GEOFARL) [reply apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts:23]: I cannot reproduce such behavior, I created notifications before applying migration, then migrated, deleted the user and notifications were gone. Can you explain why do you think the userId will be null?

File: apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { type Knex } from "knex";
+
+const TableName = {
+	CHAT_MESSAGES: "notifications",
```
Comment (fshabanov) [inline apps/backend/src/db/migrations/20240309063712_add_notifications_cascade_deletion.ts:4]:
```suggestion
	NOTIFICATIONS: "notifications",
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**